exp/internal/controllers/machinepool_controller_phases.go

```diff
@@ -283,7 +285,18 @@ func (r *MachinePoolReconciler) reconcileInfrastructure(ctx context.Context, clu
 		conditions.WithFallbackValue(ready, clusterv1.WaitingForInfrastructureFallbackReason, clusterv1.ConditionSeverityInfo, ""),
 	)
 
-	if err := r.reconcileMachines(ctx, mp, infraConfig); err != nil {
+	clusterClient, err := r.Tracker.GetClient(ctx, util.ObjectKey(cluster))
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// Get the nodeRefsMap from the cluster.
+	s.nodeRefMap, err = r.getNodeRefMap(ctx, clusterClient)
+	if err != nil {
+		r.recorder.Event(mp, corev1.EventTypeWarning, "FailedGetNodeRefMap", err.Error())
```

Comment:
I would prefer to return an error here and not send an event. Usually we're bubbling up errors and not emitting events when we hit them.

I think this could produce a lot of events when a workload cluster is not reachable

Reply:
gotcha -- right right i think i took this approach because if we return an error at this point, then we stop short of `reconcileMachines` which would a be change in behavior (previously `reconcileMachines` was called without any node references as we didn't set the status).  

can make the update to return at this point if you think that's best?  or maybe emit a `Warn` log indicating status won't be set on the machines and let execution continue?

Reply:
Maybe let's store this error here e.g. in `getNodeRefsErr` and then check in l.300+ if either this error is !=nil or the one returned from `reconcileMachines` and then return if at least one of the errors is != nil + aggregate them.

Reply:
nice yes sounds good -- just pushed up this change